activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -184,6 +177,27 @@ def ids
 
     private
 
+      def _pluck(column_names, unsafe_raw)
+        unrecognized = column_names.reject do |cn|
+          @klass.respond_to_attribute?(cn)
+        end
+
+        if loaded? && unrecognized.none?
+          records.pluck(*column_names)
+        elsif has_include?(column_names.first)
+          construct_relation_for_association_calculations.pluck(*column_names)
+        elsif unsafe_raw || unrecognized.none?
```

Comment:
Feels like this is a conditional checking the guard condition, but it's actually checking whether *not* to guard, leaving the raise in the `else` disconnected from the decisionmaking that leads to it. Compare with:
```
elsif !allow_unsafe_raw && unrecognized.any?
  raise …
else
  # actual pluck impl
end
```